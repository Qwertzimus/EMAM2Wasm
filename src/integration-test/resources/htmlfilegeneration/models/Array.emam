package models;

component Array {
  ports
    in Q(-10 : 5 ) inArray[3],
    out Q(-10/3 m/s : 10 km/h) outArray[4];
}