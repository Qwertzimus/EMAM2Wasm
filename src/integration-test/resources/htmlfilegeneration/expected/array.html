<!doctype html>
<html lang="en-us">
<style>
  .model {
    overflow: hidden;
    width: fit-content;
  }

  .ports {
    overflow: hidden;
  }

  .inports {
    float: left;
    margin: 10px;
  }

  .outports {
    float: left;
    margin: 10px;
  }

  .inport-header {
    margin: 0 auto;
    width: fit-content;
  }

  .inport-label {
    float: left;
    margin-right: 5px;
  }

  .inport-field {
    float: right;
    width: 200px;
  }

  .outport-header {
    margin: 0 auto;
    width: fit-content;
  }

  .outport-label {
    float: left;
    margin-right: 5px;
  }

  .outport-field {
    float: left;
    min-width: 200px;
  }

  .buttons {
    display: flex;
    flex-direction: row;
    margin: 10px;
  }

  .buttons div {
    flex-basis: 100%;
  }

  .button {
    color: white;
    padding: 10px 16px;
    border: none;
    text-align: center;
    text-decoration: none;
    font-size: 14px;
    cursor: pointer;
    width: 100px;
    flex-basis: 100%;
  }

  #reset {
    background-color: #d22b23;
  }

  #execute {
    background-color: #24d231;
  }

  .error {
    color: red;
    margin: 10px;
  }

  .inport-container-inArray {
    overflow: hidden;
    margin: 3px;
  }

  .outport-container-outArray {
    overflow: hidden;
    margin: 3px;
  }
</style>
<head>
  <script type="text/javascript"
          src="https://cdnjs.cloudflare.com/ajax/libs/mathjs/4.0.1/math.min.js"></script>
  <script type="text/javascript" src="array_wrapper.js"></script>
  <script type="text/javascript" src="array.js"></script>
  <title>Model</title>
</head>
<body>
<div class="container">
  <div class="model">
    <div class="ports">
      <div class="inports">
        <div class="inport-header">
          <h3>Inports</h3>
        </div>
        <div class="inport-container-inArray">
          <div class="inport-label">
            inArray:
          </div>
          <div class="inport-field">
            <input type="text" id="inport-field-inArray">
          </div>
        </div>
      </div>
      <div class="outports">
        <div class="outport-header">
          <h3>Outports</h3>
        </div>
        <div class="outport-container-outArray">
          <div class="outport-label">
            outArray:
          </div>
          <div class="outport-field" id="outport-field-outArray"></div>
        </div>
      </div>
    </div>
    <div class="buttons">
      <div class="reset">
        <button class="button" id="reset" onclick="reset()">Reset</button>
      </div>
      <div class="execute">
        <button class="button" id="execute" onclick="exec()">Execute</button>
      </div>
    </div>
  </div>
  <div class="error" id="error"></div>
</div>

<script>
    function exec() {
        clearOutportFields();
        clearErrors();

        try {
            setInArray(document.getElementById("inport-field-inArray").value);

            execute();

            document.getElementById("outport-field-outArray").innerText = getOutArray();
        }
        catch (err) {
            if (err.message === undefined) {
                document.getElementById("error").innerText = err;
            }
            else {
                document.getElementById("error").innerHTML = err.message;
            }
        }
    }

    function reset() {
        init();
        clearInportFields();
        clearOutportFields();
        clearErrors();
    }

    function clearInportFields() {
        document.getElementById("inport-field-inArray").value = "";
    }

    function clearOutportFields() {
        document.getElementById("outport-field-outArray").innerText = "";
    }

    function clearErrors() {
        document.getElementById("error").innerHTML = "";
    }
</script>
</body>
</html>
